src/Workspaces/Core/Portable/Editing/SyntaxEditor.cs

```diff
@@ -36,6 +38,30 @@ internal SyntaxEditor(SyntaxNode root, SyntaxGenerator generator)
             _changes = new List<Change>();
         }
 
+        private SyntaxNode ApplyTrackingToNewNode(SyntaxNode node)
+        {
+            if (node == null)
```

Comment:
When will this be true? Only when `ReplaceNode` is called and the modifier returns null?

Reply:
`ReplaceNode(node, newNode)` => `newNode` can be null. In fact there are [unit tests](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces.UnitTests/CodeGeneration/SymbolEditorTests.cs,900) which failed, which led me to add this null check.